samples/bluetooth/direction_finding_central/src/main.c

```diff
@@ -0,0 +1,270 @@
+			return true;
+		}
+
+		(void)memcpy(&u64, data->data, data->data_len);
+
+		u64 = sys_le64_to_cpu(u64);
+
+		if (!(u64 & DF_FEAT_ENABLED)) {
+			return true;
+		}
+
+		err = bt_le_scan_stop();
+		if (err) {
+			printk("Stop LE scan failed (err %d)\n", err);
+			return true;
+		}
+
+		err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, &conn_params, &default_conn);
+		if (err) {
+			printk("Create conn failed (err %d)\n", err);
+			start_scan();
+		}
+		return false;
+	}
+
+	return true;
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char dev[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, dev, sizeof(dev));
+	printk("[DEVICE]: %s, AD evt type %u, AD data len %u, RSSI %i\n", dev, type, ad->len, rssi);
+
+	/* We're only interested in connectable events */
+	if (type == BT_GAP_ADV_TYPE_ADV_IND || type == BT_GAP_ADV_TYPE_ADV_DIRECT_IND) {
+		bt_data_parse(ad, eir_found, (void *)addr);
+	}
+}
+
+static void enable_cte_reqest(void)
+{
+	int err;
+
+	const struct bt_df_conn_cte_rx_param cte_rx_params = {
+#if defined(CONFIG_BT_DF_CTE_RX_AOA)
+		.cte_types = BT_DF_CTE_TYPE_ALL,
+		.slot_durations = 0x2,
+		.num_ant_ids = ARRAY_SIZE(ant_patterns),
+		.ant_ids = ant_patterns,
+#else
+		.cte_types = BT_DF_CTE_TYPE_AOD_1US | BT_DF_CTE_TYPE_AOD_2US,
+#endif /* CONFIG_BT_DF_CTE_RX_AOA */
+	};
+
+	const struct bt_df_conn_cte_req_params cte_req_params = {
+		.interval = CTE_REQ_INTERVAL,
+		.cte_length = CTE_LEN,
+#if defined(CONFIG_BT_DF_CTE_RX_AOA)
+		.cte_type = BT_DF_CTE_TYPE_AOA,
+#else
+		.cte_type = BT_DF_CTE_TYPE_AOD_2US,
+#endif /* CONFIG_BT_DF_CTE_RX_AOA */
+	};
+
+	printk("Enable receiving of CTE...\n");
+	err = bt_df_conn_cte_rx_enable(default_conn, &cte_rx_params);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success. CTE receive enabled.\n");
+
+	printk("Request CTE from peer device...\n");
+	err = bt_df_conn_cte_req_enable(default_conn, &cte_req_params);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success. CTE request enabled.\n");
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	/* Use active scanning and disable duplicate filtering to handle any
+	 * devices that might update their advertising data at runtime.
+	 */
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("Failed to connect to %s (%u)\n", addr, conn_err);
+
+		bt_conn_unref(default_conn);
```

Comment:
Should we unref here?

Reply:
The reference is obtained when `bt_conn_le_create` is called in line: 103